### ISSUE TITLE
Ideas

### ISSUE THREAD

**Issue by A-Hopkins** [A-Hopkins/RocketLab#1]
A place to write down various ideas and flesh out the rocket. All ideas are up in the air (ha!).

# Hardware
## Features
* Solid Rocket Motor
* Stages
* TVC
* Actuator controlled fins
* Recovery method
  * TVC Landings
  * Parachutes
  * RC car with net
* Sensors
  * Altimeter
  * GPS
  * 
* Flight computer
  * Microcontroller (something small and very cheap)
    * Ardiuno of some flavor perhaps?

# Software
## Features
* Flight s/w can be dropped into sim or onto hardware with little to no configuration changes
* Hard Real Time events scheduling

# Flight Profile
## Options
* Single stage, parachute recovery
* Two-stage, parachute recovery
  * IF Parachute recovery, could look into catching the parachutes then
* ...

# Notes
* Need to make goals clear and concrete. I think defining the goal flight profile will help us come up with requirements/a design, but a cool(/attainable) flight profile doesn't have to be the main goal.
## Ideas for Objectives
* Unconventional flight profile
* Very conventional flight profile done well
* Maximize altitude over wet mass ratio
* Launching and hard-landing on a target
* Reuseability (Easy to swap/refill/etc and launch again)
  * This is a must for easy testing. Ideally, as easy as swapping the motor in a "Plug n' Play" way (for a solid motor)
* Multistage, 1 for up-go, 1 for landing?

**Comment by shintoo** [comment]
AFAIK, there is one brave man trying to land model rockets, and he's gotten close but hasn't succeeded. Seems to be quite the boondoggle.

**Comment by A-Hopkins** [comment]
I see, it definitely produces challenges. So, scrapping that idea, What does the TVC give us over aerodynamic design? Perhaps, in junction with the flight computer it can produce predictive landing spots? Or maybe it's just because, which I'm totally cool over-engineering things lol

**Comment by A-Hopkins** [comment]
What did you have in mind for unconventional flight profiles?

**Comment by shintoo** [comment]
> I see, it definitely produces challenges. So, scrapping that idea, What does the TVC give us over aerodynamic design? Perhaps, in junction with the flight computer it can produce predictive landing spots? Or maybe it's just because, which I'm totally cool over-engineering things lol

Not too sure what TVC gives us other than a way to demonstrate that we know how to into TVC, lol.  There are plenty of non-recovered rockets with highly capable TVC, but they're also trying to get to a specific orbit is their reason I presume. Things definitely don't need a purpose though, learning how to do the thing itself could be the purpose :') Why not even tack on some ""RCS"", lol

> What did you have in mind for unconventional flight profiles?

I haven't a clue on anything cool.

**Comment by shintoo** [comment]
I also do fancy the idea of doing something 'boring', as long as we can implement some sort of control systems. Like a single stage, single motor, parachute-recovered rocket. But is an altimeter triggering parachutes really enough? If recovering with parachutes, that would be the minimum, but perhaps we can do one or two more things as well. Like a pre-programmed flight path, which could really be just tvc/fin actuation based on altimeter readout.

**Comment by A-Hopkins** [comment]
Okay, I'm with it. We have to walk before we run lol. I would like to see for flight plans, a target goal with some % error. To give more substance to the tvc/fin system, we could use it to guide the rocket into whatever zone we're looking for. I'd like to start off really small, probably like E sized motors are minimum to have a lift off. I think E9s have a max lift off weight of ~1lb

**Comment by A-Hopkins** [comment]
Just to throw a little bit of complexity into it >.>, maybe instead of parachute recovery we do a helicopter kind of recovery. Like small helicopter blades that deploy at apogee and self spin with coming back down and has actuators on it to give some control on descent. As opposed to taking caution to the wind so to speak

**Comment by shintoo** [comment]
> As opposed to taking caution to the wind so to speak

Spectacularly done.

Helicopter idea sounds sick as heck. Let's look into it. So we want a helicopter motor on a gimbal essentially? We can look into how helicopters themselves stabilize too.

As for blade deployment, foldable blades? Are we 'descending' the whole rocket or a little helicopter payload?

Target landing 'region' sounds good, too.

**Comment by A-Hopkins** [comment]
We wouldn't need a motor for the helicopter. The blades will be in such a way that just by descending they will spin. They wouldn't spin fast enough to generate enough lift to fly, but just kind of glide down. I think probably descend the whole rocket, it might be hard to fit the things needed in a payload area.

I'm thinking blades that open...kind of like an umbrella, they flare out after the nose cone pops maybe.

**Comment by shintoo** [comment]
Would unpowered blades spinning on descent actually create some amount of lift?

**Comment by A-Hopkins** [comment]
Yah! We'd have to do some maths to figure out how large the blades need to be, but there should be plenty of lift generated for a descent that doesn't break the rocket

**Comment by shintoo** [comment]
Okay, lit. I found [this cool page](http://www.copters.com/aero/autorotation.html) with pictures that seems somewhat helpful. There's lots of info about with autorotative landings due to that being an emergency procedure in helicopter flight in the case of an engine failure. Neat!

**Comment by A-Hopkins** [comment]
I'm thinking of a mechanism that when the nosecone pops off a bar extends up top giving enough clearance for the blades to come out as well, they could catch the wind and start their rotations. Like an umbrella extender rod thing.

P.S. Helicopter root words is not heli and copter, it's actually helico, meaning spiral, and pter, with wings

**Comment by shintoo** [comment]
Umbrella extender rod sounds like it would work but may be a lot of moving parts, would it be possible to have drag on the blades pull them open, and have them lock open? So the extender rod is there, but what mechanism for unfolding the blades?

**Comment by A-Hopkins** [comment]
Yeah, that's what I was thinking as well. The drag extending the blades and we could have a pin of some kind lock them when they're fully extended. I'm thinking the extender rod can be activated when the nosecone pops off so it isn't very complicated or electronic. 

Also, we should start thinking about concrete requirements to aide the search of hardware.

TVC: 5 degrees of bilateral movement
Fins: 10 degrees of rotation

stuff like that, so we can find the best microprocessor and actuators and all that junk, then once we have that we can figure out the size of the fuselage, and motor to move it all.

**Comment by shintoo** [comment]
Sounds good. How should we go about coming up with those kinds of requirements? If we're aiming for landing in an area with a margin of error, deciding on a goal margin of error might be useful. Also, for fun, designating that area with some sort of RF beacon and having some sort of transceiver/device on the rocket for real-time navigation would be cool. This way, we're not preprogramming a landing zone, but placing a beacon, and the rocket launches vertically, and tries to land towards the beacon.

**Comment by A-Hopkins** [comment]
Wooooow. I love that idea. Basically coming up with the requirements I’m thinking maybe look into some other peoples projects to get an idea of what is possible and then state that and that’ll provide us top of what kind of equipment we need which will in turn tell us what size fuselage and weight needs for the motors.

**Comment by shintoo** [comment]
So a lot (all) of the helicopter recovery rockets I have seen have had their helicopter module attached by cable to the actual rocket, rather than a rod extending out of the rocket itself. To control the fall of the rocket, I think it would have to be directly attached via the rod, right? Curious why a rod hasn't been done (as far as I have seen).

**Comment by A-Hopkins** [comment]
[cid:FEEEAA9F-0B15-49F3-841E-75479A2D6472-L0-001]

Okay, so the idea is a few copter blades that pivot at where it connects to the fuselage. They lay flat up to the body and the nose cone caps over them to hold them in place. When the cone pops off they bloom out to take hold.

**Comment by shintoo** [comment]
Ah I see. This way the entire rocket would have to spin though, no?

This does simplify the whole copter mechanism a lot though, and deletes a lot of parts. Very neat.

**Comment by A-Hopkins** [comment]
Yessir

On Oct 9, 2019, at 12:44, Sean Rapp <notifications@github.com> wrote:

﻿

Ah I see. This way the entire rocket would have to spin though, no?

**Comment by shintoo** [comment]
Would the rocket spinning so quickly not pose any problems? Seems like a big thing to impose on the rocket trying to land somewhere. And how would gimballing work if the copter blades are attached to the rocket?

(Or is the copter bit not intended to gimbal, and the direction to be done by fins? :-))

**Comment by A-Hopkins** [comment]
A bracket that can move is attached to some stepper motor or something that can spin both ways actually. Will be able to change the pitch of the blades and give a flight profile. I’m not sure if spinning the rocket as a whole is a good idea or not lol. I don’t think it’ll change anything by a lot. But it could be bad 🤷‍♂️

On Oct 9, 2019, at 13:17, Sean Rapp <notifications@github.com> wrote:

﻿

Would the rocket spinning so quickly not pose any problems? Seems like a big thing to impose on the rocket trying to land somewhere. And how would gimballing work if the copter blades are attached to the rocket?

**Comment by shintoo** [comment]
We can have a circular bracket that can rotate around the fuselage that the blades are attached to is what you're saying, yeah? That sounds perfect. If the bracket is extended out from the fuselage by any significant amount too, we can tilt it somehow. That part sounds tricky.

**Comment by A-Hopkins** [comment]
I like the idea of a ring, but what I had originally in mind was probably some small bracket specifically for folding that was attached to something that went into the fuselage that could rotate. But with a ring, it could be an idea to have the the blades free from the rocket so only the blade spin instead of the whole rocket. Just not sure how to move them with motors to have them move and get the wires tangled

On Oct 9, 2019, at 14:28, Sean Rapp <notifications@github.com> wrote:

﻿

We can have a circular bracket that can rotate around the fuselage that the blades are attached to is what you're saying, yeah? That sounds perfect. If the bracket is extended out from the fuselage by any significant amount too, we can tilt it somehow. That part sounds tricky.

**Comment by shintoo** [comment]
Yeah, there would have to be two layers of the ring, one that is mounted to the rocket by whatever controls are used to tilt the ring structure, and another outer layer that can rotate freely, to which the blades will be attached. Probably bearings between the two or something like that. That sounds overly complicated, and the mechanism for tilting the ring would add complexity, too; all I can think of for that are 2 or 4 servos that push and pull the ring, for 2 at 0 and 90 degrees, for 4 just 0,90,180,270. That's a lot of moving parts, though. (But no rotating wire problems etc.)

In doodle form:

https://i.imgur.com/BnwXoRg.jpg

**Comment by A-Hopkins** [comment]
Yeah that is pretty complicated. You’re right though those wires are going to be a real pain to deal with in any kind of design. Maybe the helicopter descent with control is outside the scope of this project?

On Oct 9, 2019, at 15:17, Sean Rapp <notifications@github.com> wrote:

﻿

Yeah, there would have to be two layers of the ring, one that is mounted to the rocket by whatever controls are used to tilt the ring structure, and another outer layer that can rotate freely, to which the blades will be attached. Probably bearings between the two or something like that. That sounds overly complicated, and the mechanism for tilting the ring would add complexity, too; all I can think of for that are 2 or 4 servos that push and pull the ring, for 2 at 0 and 90 degrees, for 4 just 0,90,180,270. That's a lot of moving parts, though. (But no rotating wire problems etc.)

In doodle form:

https://i.imgur.com/BnwXoRg.jpg<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fi.imgur.com%2FBnwXoRg.jpg&data=02%7C01%7Chopkinsalex%40knights.ucf.edu%7Ccdcd7c98ab94464eb65108d74ced5d5c%7C5b16e18278b3412c919668342689eeb7%7C0%7C0%7C637062454667654944&sdata=gzP9S9%2FbHHimuvG2oNUaLUq%2Fj4KZtiwmYPJb5eXjn7g%3D&reserved=0>

**Comment by shintoo** [comment]
Oh nooooooo, that was the cool part! Controlled descent via helicoptering does seem to be really complicated though. What other way could we control trajectory to land in a desired landing area? Paraglide-esque shoot? Fins alone wouldn't slow the rocket enough, I think. Maybe. Perhaps guide rocket to area with fins, and secondary engine fires just before landing to slow it down?

We don't have to rule out the copter method so quickly, though - we can just rule out using the ring/mounting the blades to the fuselage. Umbrellarodcopter is still on the table for me. (The rod can be gimballed, blades at end of rod)

**Comment by A-Hopkins** [comment]
You’re right it would be really awesome. I think the only way to get it done right is to make the full system that helicopters have to do which maybe we can do crudely that gets it done but will be heavy.

Other methods for controlled descent... we could have streamers that act almost like drogue shoots and fins that we can control. Maybe grid fins 😏

On Oct 9, 2019, at 15:41, Sean Rapp <notifications@github.com> wrote:

﻿

Oh nooooooo, that was the cool part! Controlled descent via helicoptering does seem to be really complicated though. What other way could we control trajectory to land in a desired landing area? Paraglide-esque shoot? Fins alone wouldn't slow the rocket enough, I think.

**Comment by shintoo** [comment]
any thoughts on fins for guiding and second engine for landing burn? i
don't think github emails when comments are edited, and I'm always quick to
hit the post comment button 😅

**Comment by A-Hopkins** [comment]
Wow. I didn’t realize the email responses were such aides. Fins and a second engine For landing would be pretty cool. I’m not opposed to it. We also could revisit the umbrella blades idea. I liked that one too, but we’d still have to have some way to spin and have pitch control. That system probably is inherently easier to implement though

**Comment by shintoo** [comment]
> That system probably is inherently easier to implement though

Which one are you referring to here, umbrella or landing engine?

I don't think it'd be too complicated to control the copter, but blade deployment + the control mechanism together could get a bit out of hand.

I'm feeling pretty sold on fins and landing engine, go figure xD Would drogue streamers + fins make sense? For landing engine, we'd need a way to ensure rocket is (near) vertical before landing; streamers would help with that but I'm thinking that just aerodynamically the rocket will be straight anyway. But perhaps streamers for slowing down makes sense too. (Maybe this depends on how high the rocket gets + how powerful the landing engine would be?

**Comment by A-Hopkins** [comment]
The umbrella copter is probably easier than landing engine. BUT I love a good challenge so I’m definitely down for landing engine. It takes out more of the engineering challenge and puts it more into software I think. 

Going forward with the landing engine idea, the rocket will need some deployable fins on the nose area to keep it aerodynamically stable on the way down. OR a few good streamers I think will work...

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by shintoo** [comment]
I was thinking, our two options with landing engine method are that either it comes down the same way it went up, thus needed nose flaps, *or* the rocket flips at apogee, and the landing engine is in the nose facing outward. Would be super wacky/fun, lol. The rocket might flip by itself if there are fins at the bottom and none at the top, perhaps.

**Comment by A-Hopkins** [comment]
Oh that is super wacky lol. I like it. If we do that method we’d need to be sure to have fins at the bottom part of the rocket to keep it in a downward trajectory. TVC on the landing engine top of course and we’d have no other means of slowing down potentially

**Comment by shintoo** [comment]
The fun part about the landing engine is timing the firing. How do we know when we're close to the ground? The radio meme? Altimeter? GPS?

**Comment by A-Hopkins** [comment]
Probably altimeter. Seems the cheapest/easiest?

**Comment by shintoo** [comment]
Have a bit of an idea for something; making a test stand with a force sensor to measure thrust curves of motors. Not that they're unknown, but just a small project for fun.

**Comment by A-Hopkins** [comment]
Oh yeah we should do that. It’ll be a good place to test equipment too. Plus I just love data
